src/js/components/PodDetail.js

```diff
@@ -0,0 +1,234 @@
+          'serviceRestartSuccess'
+        ]
+      }
+    ];
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  componentWillMount() {
+    if (this.props.mode === 'pod') {
+
+    }
+  }
+
+  componentDidMount() {
+    super.componentDidMount(...arguments);
+  }
+
+  componentWillUpdate() {
+    super.componentWillUpdate(...arguments);
+  }
+
+  onActionsItemSelection(item) {
+    this.setState({serviceActionDialog: item.id});
+  }
+
+  onAcceptRestartConfirmDialog() {
+    this.setState({disabledDialog: ServiceActionItem.RESTART}, () => {
+      MarathonStore.restartService(
+        this.props.pod.id, this.shouldForceUpdate(this.state.errorMsg)
+      );
+    });
+  }
+
+  onServiceDestroyModalClose() {
+    this.closeDialog();
+    this.context.router.transitionTo('services-page');
+  }
+
+  onServiceSuspendModalClose() {
+    this.closeDialog();
+  }
+
+  onMarathonStoreServiceRestartSuccess() {
+    this.closeDialog();
+  }
+
+  onMarathonStoreServiceRestartError({message:errorMsg}) {
+    this.setState({
+      disabledDialog: null,
+      errorMsg
+    });
+  }
+
+  closeDialog() {
+    this.setState({
+      disabledDialog: null,
+      errorMsg: null,
+      serviceActionDialog: null
+    });
+  }
+
+  getRestartConfirmDialog() {
+    const {pod} = this.props;
+    const {state} = this;
+
+    return (
+      <Confirm
+        disabled={state.disabledDialog === ServiceActionItem.RESTART}
+        open={state.serviceActionDialog === ServiceActionItem.RESTART}
+        onClose={this.closeDialog}
+        leftButtonText="Cancel"
+        leftButtonCallback={this.closeDialog}
+        rightButtonText="Restart Service"
+        rightButtonClassName="button button-danger"
+        rightButtonCallback={this.onAcceptRestartConfirmDialog}>
+        <div className="container-pod flush-top container-pod-short-bottom">
+          <h2 className="text-danger text-align-center flush-top">Restart Service</h2>
+          <p>
+            Are you sure you want to restart <span className="emphasize">{pod.getId()}</span>?
+          </p>
+          {this.getErrorMessage()}
+        </div>
+      </Confirm>
+    );
+  }
+
+  shouldForceUpdate(message = this.state.errorMsg) {
+    return message && /force=true/.test(message);
+  }
+
+  getErrorMessage() {
+    let {errorMsg} = this.state;
+    if (!errorMsg) {
+      return null;
+    }
+
+    if (this.shouldForceUpdate(errorMsg)) {
+      return (
+        <h4 className="text-align-center text-danger flush-top">
+          App is currently locked by one or more deployments. Press the button
+          again to forcefully change and deploy the new configuration.
+        </h4>
+      );
+    }
+
+    return (
+      <p className="text-danger flush-top">{errorMsg}</p>
+    );
+  }
+
+  renderConfigurationTabView() {
+    return (
+      <ServiceDetailConfigurationTab service={this.props.pod} />
+    );
+  }
+
+  renderDebugTabView() {
```

Comment:
Thi is sad. Creating more tech debt in stead of doing it up in front. We have tasks to pay these down, but this will only get bigger.
I am happy with separate PRs as long as it gets handle with this task and not another task